packages/checkout/widgets-lib/src/widgets/bridge/components/BridgeReviewSummary.tsx

```diff
@@ -189,9 +195,50 @@ export function BridgeReviewSummary() {
     })();
   }, []);
 
+  const handleNetworkSwitch = useCallback((provider: Web3Provider) => {
+    bridgeDispatch({
+      payload: {
+        type: BridgeActions.SET_WALLETS_AND_NETWORKS,
+        from: {
+          web3Provider: provider,
+          walletAddress: from?.walletAddress!,
+          network: from?.network!,
+        },
+        to: {
+          web3Provider: to?.web3Provider!,
+          walletAddress: to?.walletAddress!,
+          network: to?.network!,
+        },
+      },
+    });
+  }, [from?.web3Provider, from?.network, to?.web3Provider, to?.network]);
+
+  useEffect(() => {
+    if (!from?.web3Provider) return () => {};
+
+    const handleChainChanged = () => {
+      const newProvider = new Web3Provider(from?.web3Provider.provider);
+      handleNetworkSwitch(newProvider);
+      setShowSwitchNetworkDrawer(false);
+    };
+    addChainChangedListener(from?.web3Provider, handleChainChanged);
+
+    return () => {
+      removeChainChangedListener(from?.web3Provider, handleChainChanged);
+    };
+  }, [from?.web3Provider]);
+
   const submitBridge = useCallback(async () => {
     if (!approveTransaction || !transaction) return;
 
+    const currentChainId = await (from?.web3Provider.provider as any).request({ method: 'eth_chainId', params: [] });
```

Comment:
Good place for a try/catch I think, just wondering what happens if the currentChainId fails? or not supported?

Reply:
Good point, will add that in